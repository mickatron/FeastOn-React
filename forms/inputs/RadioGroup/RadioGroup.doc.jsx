/* eslint-disable */
import React from 'react';
import Component from './RadioGroup';
import PropTypes from 'prop-types';

/**
 * A RadioGroup is a collection of HTML radio's rendered as a group,
 * when grouped only one option in the group can be selected.
 *
 * @composedWith UiWrapper
*/
function RadioGroup(props) {
  return <Component {...props} />;
}

RadioGroup.propTypes = {
  /** A unique name for the radio controls to be grouped by. */
  name: PropTypes.string.isRequired,
  /** An array of Objects containing props suitable for rendering a radio. */
  radios: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
}

export default RadioGroup;
