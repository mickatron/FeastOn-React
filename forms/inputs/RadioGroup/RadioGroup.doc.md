```js
<RadioGroup
  name="RadioGroup"
  radios={[
    {
      label: 'A',
      id: 'A',
      value: 'A',
    },{
      label: 'B',
      id: 'B',
      value: 'B',
    },{
      label: 'C',
      id: 'C',
      value: 'C',
    },
  ]}
/>

```