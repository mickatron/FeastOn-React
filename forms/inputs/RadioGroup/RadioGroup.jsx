import PropTypes from 'prop-types';
import React from 'react';

import Label from '../../Label/Label';
import uiWrapper from '../../../HOC/uiWrapper';

import HtmlInput from '../HtmlInput/HtmlInput';

class RadioGroup extends React.PureComponent {
  static propTypes = {
    /** A unique name for the radio controls to be grouped by. */
    name: PropTypes.string.isRequired,
    /** An array of Objects containing props suitable for rendering a radio. */
    // TODO: should this be renderProps, and an array of renderProps?
    radios: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })).isRequired,
  }

  render() {
    const {
      name,
      radios,
    } = this.props;

    return (
      <React.Fragment>
        {radios.map(radio => (
          <Label
            key={radio.value}
            text={radio.label}
            renderInput="left"
            input={({ uid }) => (
              <HtmlInput
                {...radio}
                type="radio"
                name={name}
                id={uid}
              />
            )}
          />
        ))}
      </React.Fragment>
    );
  }
}

export default uiWrapper({
  name: 'RadioGroup',
  element: 'div',
  Component: RadioGroup,
});
