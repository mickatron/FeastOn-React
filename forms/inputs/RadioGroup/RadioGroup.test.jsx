import { render } from 'enzyme';
import React from 'react';
import RadioGroup from './RadioGroup';

describe('RadioGroup', () => {
  let component = null;

  it('renders', () => {
    component = render((
      <RadioGroup
        name="RadioGroup"
        radios={[
          {
            label: 'A',
            id: 'A',
            value: 'A',
          }, {
            label: 'B',
            id: 'B',
            value: 'B',
          }, {
            label: 'C',
            id: 'C',
            value: 'C',
          },
        ]}
      />
    ));
    expect(component).toMatchSnapshot();
  });
});
