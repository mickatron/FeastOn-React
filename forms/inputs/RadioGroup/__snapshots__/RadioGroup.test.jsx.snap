// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "RadioGroup",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Label",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Input Input-Radio",
              "id": "1",
              "label": "A",
              "name": "RadioGroup",
              "type": "radio",
              "value": "A",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "f-Label__text",
              },
              "children": Array [
                Object {
                  "data": "A",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
              "label": undefined,
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
              "label": undefined,
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "f-Label__text",
            },
            "children": Array [
              Object {
                "data": "A",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "Input Input-Radio",
                "id": "1",
                "label": "A",
                "name": "RadioGroup",
                "type": "radio",
                "value": "A",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "Label",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Input Input-Radio",
                "id": "2",
                "label": "B",
                "name": "RadioGroup",
                "type": "radio",
                "value": "B",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "f-Label__text",
                },
                "children": Array [
                  Object {
                    "data": "B",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "f-Label__text",
              },
              "children": Array [
                Object {
                  "data": "B",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "Input Input-Radio",
                  "id": "2",
                  "label": "B",
                  "name": "RadioGroup",
                  "type": "radio",
                  "value": "B",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "Label",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Input Input-Radio",
                  "id": "3",
                  "label": "C",
                  "name": "RadioGroup",
                  "type": "radio",
                  "value": "C",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "f-Label__text",
                  },
                  "children": Array [
                    Object {
                      "data": "C",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "f-Label__text",
                },
                "children": Array [
                  Object {
                    "data": "C",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "Input Input-Radio",
                    "id": "3",
                    "label": "C",
                    "name": "RadioGroup",
                    "type": "radio",
                    "value": "C",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                    "label": undefined,
                    "name": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                    "label": undefined,
                    "name": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "Label",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Input Input-Radio",
              "id": "2",
              "label": "B",
              "name": "RadioGroup",
              "type": "radio",
              "value": "B",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "f-Label__text",
              },
              "children": Array [
                Object {
                  "data": "B",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
              "label": undefined,
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
              "label": undefined,
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "f-Label__text",
            },
            "children": Array [
              Object {
                "data": "B",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "Input Input-Radio",
                "id": "2",
                "label": "B",
                "name": "RadioGroup",
                "type": "radio",
                "value": "B",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "Label",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Input Input-Radio",
                "id": "3",
                "label": "C",
                "name": "RadioGroup",
                "type": "radio",
                "value": "C",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "f-Label__text",
                },
                "children": Array [
                  Object {
                    "data": "C",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "f-Label__text",
              },
              "children": Array [
                Object {
                  "data": "C",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "Input Input-Radio",
                  "id": "3",
                  "label": "C",
                  "name": "RadioGroup",
                  "type": "radio",
                  "value": "C",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "Label",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Input Input-Radio",
                "id": "1",
                "label": "A",
                "name": "RadioGroup",
                "type": "radio",
                "value": "A",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "f-Label__text",
                },
                "children": Array [
                  Object {
                    "data": "A",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "f-Label__text",
              },
              "children": Array [
                Object {
                  "data": "A",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "Input Input-Radio",
                  "id": "1",
                  "label": "A",
                  "name": "RadioGroup",
                  "type": "radio",
                  "value": "A",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "Label",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Input Input-Radio",
              "id": "3",
              "label": "C",
              "name": "RadioGroup",
              "type": "radio",
              "value": "C",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "f-Label__text",
              },
              "children": Array [
                Object {
                  "data": "C",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
              "label": undefined,
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
              "label": undefined,
              "name": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "f-Label__text",
            },
            "children": Array [
              Object {
                "data": "C",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "Input Input-Radio",
                "id": "3",
                "label": "C",
                "name": "RadioGroup",
                "type": "radio",
                "value": "C",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "Label",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "Input Input-Radio",
                "id": "2",
                "label": "B",
                "name": "RadioGroup",
                "type": "radio",
                "value": "B",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "f-Label__text",
                },
                "children": Array [
                  Object {
                    "data": "B",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "label": undefined,
                "name": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            Object {
              "attribs": Object {
                "class": "f-Label__text",
              },
              "children": Array [
                Object {
                  "data": "B",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "Input Input-Radio",
                  "id": "2",
                  "label": "B",
                  "name": "RadioGroup",
                  "type": "radio",
                  "value": "B",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "Label",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "Input Input-Radio",
                  "id": "1",
                  "label": "A",
                  "name": "RadioGroup",
                  "type": "radio",
                  "value": "A",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "f-Label__text",
                  },
                  "children": Array [
                    Object {
                      "data": "A",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "label": undefined,
                  "name": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "f-Label__text",
                },
                "children": Array [
                  Object {
                    "data": "A",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "Input Input-Radio",
                    "id": "1",
                    "label": "A",
                    "name": "RadioGroup",
                    "type": "radio",
                    "value": "A",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                    "label": undefined,
                    "name": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                    "label": undefined,
                    "name": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
